options.js

```diff
@@ -88,88 +94,123 @@ function updateShortcutInputText(inputId, keyCode) {
+    '<option value="0">Do not disable website key bindings</option>' +
+    '<option value="1">Disable websites key bindings</option>' +
+    '</select>' +
+    '<button class="removeParent">X</button>';
+  var div = document.createElement('div');
+  div.setAttribute('class', 'row customs');
+  div.innerHTML = html;
+  var customs_element = document.getElementById("customs");
+  customs_element.insertBefore(div, customs_element.children[customs_element.childElementCount - 1]);
+}
+
+function createKeyBindings() {
+  var doo = $(this).find(".customDo").val();
+  var key = $(this).find(".customKey")[0].keyCode;
+  var val = Number($(this).find(".customValue").val());
+  var ret = Number($(this).find(".customReturn").val());
+
+  keyBindings.push([doo, key, val, ret]);
+}
+
 // Saves options to chrome.storage
 function save_options() {
+  keyBindings = [];
+  $(".customs").each(createKeyBindings)
 
-  var speedStep     = document.getElementById('speedStep').value;
-  var rewindTime    = document.getElementById('rewindTime').value;
-  var advanceTime   = document.getElementById('advanceTime').value;
-  var fastSpeed     = document.getElementById('fastSpeed').value;
-  var resetKeyCode  = document.getElementById('resetKeyInput').keyCode;
-  var rewindKeyCode = document.getElementById('rewindKeyInput').keyCode;
-  var advanceKeyCode = document.getElementById('advanceKeyInput').keyCode;
-  var slowerKeyCode = document.getElementById('slowerKeyInput').keyCode;
-  var fasterKeyCode = document.getElementById('fasterKeyInput').keyCode;
-  var fastKeyCode   = document.getElementById('fastKeyInput').keyCode;
   var displayKeyCode = document.getElementById('displayKeyInput').keyCode;
   var rememberSpeed = document.getElementById('rememberSpeed').checked;
   var startHidden = document.getElementById('startHidden').checked;
-  var blacklist     = document.getElementById('blacklist').value;
-
-  speedStep     = isNaN(speedStep) ? tcDefaults.speedStep : Number(speedStep);
-  rewindTime    = isNaN(rewindTime) ? tcDefaults.rewindTime : Number(rewindTime);
-  advanceTime   = isNaN(advanceTime) ? tcDefaults.advanceTime : Number(advanceTime);
-  fastSpeed     = isNaN(fastSpeed) ? tcDefaults.fastSpeed : Number(fastSpeed);
-  resetKeyCode  = isNaN(resetKeyCode) ? tcDefaults.resetKeyCode : resetKeyCode;
-  rewindKeyCode = isNaN(rewindKeyCode) ? tcDefaults.rewindKeyCode : rewindKeyCode;
-  advanceKeyCode = isNaN(advanceKeyCode) ? tcDefaults.advanceKeyCode : advanceKeyCode;
-  slowerKeyCode = isNaN(slowerKeyCode) ? tcDefaults.slowerKeyCode : slowerKeyCode;
-  fasterKeyCode = isNaN(fasterKeyCode) ? tcDefaults.fasterKeyCode : fasterKeyCode;
-  fastKeyCode   = isNaN(fastKeyCode) ? tcDefaults.fastKeyCode : fastKeyCode;
+  var blacklist = document.getElementById('blacklist').value;
+  var version = tcDefaults.version;
+
   displayKeyCode = isNaN(displayKeyCode) ? tcDefaults.displayKeyCode : displayKeyCode;
 
+  chrome.storage.sync.clear();
```

Comment:
Why do we need the clear()?

Reply:
Can we make this conditional? As in, I do see the utility of having a ~migrate function between versions, but I don't think we need to clear this on every update? 

E.g. if previously persisted version != current version, run clear before persisting new set.